src/deepmol/evaluator/evaluator.py

```diff
@@ -129,29 +89,23 @@ def compute_model_performance(self,
         all_task_scores: dict
             If `per_task_metrics == True`, then returns a second dictionary of scores for each task separately.
         """
-        # Process input metrics
-        metrics = _process_metric_input(metrics)
-
+        n_tasks = self.dataset.n_tasks
         y = self.dataset.y
-
         y_pred = self.model.predict(self.dataset)
-        n_tasks = self.dataset.n_tasks
+        if not y.shape == np.array(y_pred).shape:
+            y_pred = normalize_labels_shape(y_pred, n_tasks)
 
         multitask_scores = {}
         all_task_scores = {}
 
         # Compute multitask metrics
         for metric in metrics:
-            results = metric.compute_metric(y,
-                                            y_pred,
-                                            per_task_metrics=per_task_metrics,
-                                            n_tasks=n_tasks,
-                                            n_classes=n_classes)
+            results = metric.compute_metric(y, y_pred, per_task_metrics=per_task_metrics, n_tasks=n_tasks)
             if per_task_metrics:
                 multitask_scores[metric.name], computed_metrics = results
                 all_task_scores[metric.name] = computed_metrics
             else:
-                multitask_scores[metric.name] = results
+                multitask_scores[metric.name], _ = results
 
         if not per_task_metrics:
             return multitask_scores, None
```

Comment:
why not return **all_task_scores** here? of course it will go empty but at least we know that is the variable **all_task_scores** and not a NoneType. Idk, just asking :)

Reply:
Yeah, makes sense.